nixos/modules/services/networking/yggdrasil.nix

```diff
@@ -0,0 +1,119 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.yggdrasil;
+in {
+  options = with types; {
+    services.yggdrasil = {
+      enable = mkEnableOption "the yggdrasil system service";
+
+      configFile = mkOption {
+        type =  nullOr str;
+        default = null;
+        example = "/run/keys/yggdrasil.conf";
+        description = ''
+          A file which contains JSON or HJSON configuration for
+          yggdrasil.
+
+          This does not have to be a complete configuration, as
+          yggdrasil will use default values for anything which is
+          omitted.  If the encryption and signing keys are omitted,
+          yggdrasil will generate new ones each time the service is
+          started, resulting in a random IPv6 address on the yggdrasil
+          network each time.
+
+          If this option is set, <option>config</option> will be ignored.
+        '';
+      };
+
+      config = mkOption {
+        type = str;
+        default = "";
+        description = ''
```

Comment:
would an example config would be appropriate? or is it to long to be inline here? maybe a minimal/partial one

Reply:
What do you think of adding to the description a suggestion to use `nix-shell -p yggdrasil --run "yggdrasil -genconf"` to see what the available options are?